WordPressKit/ReaderPostServiceRemote+Cards.swift

```diff
@@ -29,45 +29,94 @@ extension ReaderPostServiceRemote {
+        let path = "read/tags/cards"
+        guard let requestUrl = cardsEndpoint(with: path,
+                                             topics: topics,
+                                             page: page,
+                                             sortingOption: sortingOption,
+                                             refreshCount: refreshCount) else {
             return
         }
+        fetch(requestUrl, success: success, failure: failure)
+    }
+
+    /// Returns a collection of RemoteReaderCard using the discover streams API
+    /// a Reader Card can represent an item for the reader feed, such as
+    /// - Reader Post
+    /// - Topics you may like
+    /// - Blogs you may like and so on
+    ///
+    /// - Parameter topics: an array of String representing the topics
+    /// - Parameter page: a String that represents a page handle
+    /// - Parameter sortingOption: a ReaderSortingOption that represents a sorting option
+    /// - Parameter count: the number of cards to fetch. Warning: This also changes the number of objects returned for recommended sites/tags.
+    /// - Parameter success: Called when the request succeeds and the data returned is valid
+    /// - Parameter failure: Called if the request fails for any reason, or the response data is invalid
+    public func fetchStreamCards(for topics: [String],
+                                 page: String? = nil,
+                                 sortingOption: ReaderSortingOption = .noSorting,
+                                 refreshCount: Int? = nil,
+                                 count: Int? = nil,
+                                 success: @escaping ([RemoteReaderCard], String?) -> Void,
+                                 failure: @escaping (Error) -> Void) {
+        let path = "read/streams/discover"
+        guard let requestUrl = cardsEndpoint(with: path,
+                                             topics: topics,
+                                             page: page,
+                                             sortingOption: sortingOption,
+                                             count: count,
+                                             refreshCount: refreshCount) else {
+            return
+        }
+        fetch(requestUrl, success: success, failure: failure)
+    }
 
-        wordPressComRestApi.GET(requestUrl,
+    private func fetch(_ endpoint: String,
+                       success: @escaping ([RemoteReaderCard], String?) -> Void,
+                       failure: @escaping (Error) -> Void) {
+        wordPressComRestApi.GET(endpoint,
                                 parameters: nil,
                                 success: { response, _ in
+            do {
+                let decoder = JSONDecoder()
+                let data = try JSONSerialization.data(withJSONObject: response, options: [])
+                let envelope = try decoder.decode(ReaderCardEnvelope.self, from: data)
```

Comment:
BTW, you can avoid the redundant JSON encoding and decoding by using the new `perform(..., type: <Decodable>) API in `WordPressComRestApi`. Here is an example:

https://github.com/wordpress-mobile/WordPressKit-iOS/blob/94fca758e2f1b27fe655b7bb359c960eca1c6da2/WordPressKit/AnnouncementServiceRemote.swift#L17-L23

Reply:
Thanks! I've updated it to use this method in: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/744/commits/8b8d2097edb6f19568978807cfdbef10c3e7c184